The startup ecosystem is a dynamic and rapidly evolving space, with certain industries and regions attracting more investment than others. This project aims to analyze startup growth and investment trends to identify the industries that receive the most funding, the countries with the fastest-growing startups, and the relationship between funding rounds and growth rates.

To achieve this, I will be using a dataset from Kaggle that includes key variables such as industry, funding rounds, investment amounts, valuation, number of investors, country, year founded, and growth rate. However, before conducting any analysis, it is crucial to clean the dataset to handle missing values, inconsistencies, and potential outliers. Proper data cleaning ensures the accuracy and reliability of insights, allowing for well-informed conclusions about investment opportunities and growth trends in the startup world.

In [ ]:
# upload file into colab
from google.colab import files
uploaded = files.upload()

Saving startup_growth_investment_data.csv to startup_growth_investment_data.csv


In [ ]:
# load the "startup_growth_investment_data.csv" as a DataFrame
import pandas as pd
df = pd.read_csv('startup_growth_investment_data.csv')

In [ ]:
# show first rows of df
df.head()

Startup Name    Industry  Funding Rounds  Investment Amount (USD)  \
0    Startup_1  Blockchain               8             1.335166e+09   
1    Startup_2        SaaS               2             2.781498e+09   
2    Startup_3      EdTech              10             3.309032e+09   
3    Startup_4     Fintech               5             4.050196e+09   
4    Startup_5      EdTech               9             1.645080e+09   

   Valuation (USD)  Number of Investors    Country  Year Founded  \
0     6.621448e+09                   50    Germany          2012   
1     8.363214e+09                   36         UK          2006   
2     1.548270e+10                   39  Singapore          2016   
3     1.268253e+10                   44     France          2021   
4     6.887966e+09                   48      India          2011   

   Growth Rate (%)  
0            77.10  
1           105.52  
2           190.47  
3             9.44  
4           192.00

In [ ]:
# show info on df
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 9 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Startup Name             5000 non-null   object 
 1   Industry                 5000 non-null   object 
 2   Funding Rounds           5000 non-null   int64  
 3   Investment Amount (USD)  5000 non-null   float64
 4   Valuation (USD)          5000 non-null   float64
 5   Number of Investors      5000 non-null   int64  
 6   Country                  5000 non-null   object 
 7   Year Founded             5000 non-null   int64  
 8   Growth Rate (%)          5000 non-null   float64
dtypes: float64(3), int64(3), object(3)
memory usage: 351.7+ KB


In [ ]:
# describe df
df.describe()

Funding Rounds  Investment Amount (USD)  Valuation (USD)  \
count     5000.000000             5.000000e+03     5.000000e+03   
mean         5.491600             2.455567e+09     7.971059e+09   
std          2.913353             1.423787e+09     5.479487e+09   
min          1.000000             1.102610e+06     1.722547e+06   
25%          3.000000             1.221506e+09     3.598305e+09   
50%          6.000000             2.460634e+09     7.002304e+09   
75%          8.000000             3.639951e+09     1.147676e+10   
max         10.000000             4.999544e+09     2.470906e+10   

       Number of Investors  Year Founded  Growth Rate (%)  
count          5000.000000   5000.000000      5000.000000  
mean             25.542000   2011.544000       102.091732  
std              14.271838      6.885285        56.179781  
min               1.000000   2000.000000         5.060000  
25%              13.000000   2006.000000        52.815000  
50%              25.000000   2012.000000       102.215000  
75%              38.000000   2018.000000       150.580000  
max              50.000000   2023.000000       199.970000

I have imported the data into colab, and displayed some basic information to make sure the data is ready to be analyzed. There appears to be no missing values, and all data types are appropriate for the columns. Therefore, we can proceed.

In [ ]:
# identify missing values in df
df.isnull().sum()

Startup Name               0
Industry                   0
Funding Rounds             0
Investment Amount (USD)    0
Valuation (USD)            0
Number of Investors        0
Country                    0
Year Founded               0
Growth Rate (%)            0
dtype: int64

There are no missing values nor incorrect data types in this dataset

In [ ]:
# identify duplicate values in df
df.duplicated().sum()

0

There are no duplicate values in this dataset

In [ ]:
# import plotly
import plotly.express as px

# create a boxplot for the "Growth Rate" column
fig = px.box(df, y='Growth Rate (%)')

# show the plot
fig.show()

# create another bloxplot for "Investment Amount (USD)"
fig = px.box(df, y='Investment Amount (USD)')

# show the plot
fig.show()

There are no clear outliers in either of the "growth rate" or "investment amount" columns

In [ ]:
# identify values in "Number of Investors" column below or euqal to 0
df[df['Number of Investors'] <= 0]

Empty DataFrame
Columns: [Startup Name, Industry, Funding Rounds, Investment Amount (USD), Valuation (USD), Number of Investors, Country, Year Founded, Growth Rate (%)]
Index: []

In [ ]:
# What is the valuation for startups?



# create a histogram for the "Valuation (USD)"
fig = px.histogram(df, x = 'Valuation (USD)')

# show the plot
fig.show()

This histogram shows the count of startups in each valuation bin. The histogram is right-skewed, meaning that most startups are valuated below $15 billion. Some startups, however, are evaluated at 20 billion or more.

The descriptioin of the data from earlier shows that the average valuation is approximately 7.97 Billion, which is on left-most side of the chart.

In [ ]:
# create the csv file
df.to_csv('startup_growth_investment_data_cleaned.csv', index=False)

In [ ]:
# download csv file
files.download('startup_growth_investment_data_cleaned.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

This new cleaned dataset ensures that the analysis is accurate, reliable, and free from biases caused by missing or inconsistent data. By removing errors and standardizing values, the dataset becomes more suitable for identifying meaningful trends in startup growth and investment. This improves the quality of insights, allowing for better decision-making regarding industry opportunities and investment strategies.